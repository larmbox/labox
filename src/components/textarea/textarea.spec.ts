import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LTextarea } from '~/index';

test('mount component', () => {
  expect(LTextarea).toBeTruthy();

  const component = mount(LTextarea, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('textarea');
  expect(component.html()).toContain('form-field');
});

test('register css variables', () => {
  mount(LTextarea);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-textarea');
});

test('inherit attributes', () => {
  const component = mount(LTextarea, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
