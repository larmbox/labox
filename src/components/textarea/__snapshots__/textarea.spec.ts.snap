// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"form-control form-field textarea textarea-md textarea-primary\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"textarea-inner\\"><textarea id=\\"id\\" style=\\"resize: both;\\"></textarea></div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
